web3/contract.py

```diff
@@ -764,9 +765,48 @@ def __prepared_function(self, **kwargs):
                     "The only allowed keyword arguments are: %s" % self.ALLOWED_MODIFIERS)
         else:
             raise TypeError("Use up to one keyword argument, one of: %s" % self.ALLOWED_MODIFIERS)
-        contract_modifier_func = getattr(self.__contract, modifier)
+        contract_modifier_func = getattr(self._contract, modifier)
         return getattr(contract_modifier_func(modifier_dict), self.__function)
 
+class ImplicitContract(ConciseContract):
+    '''
+    ImplicitContract class is similar to the ConciseContract class
+    however it performs a transaction instead of a call if no modifier
+    is given and the method is not marked 'constant' in the ABI.
+
+    The transaction will use the default account and does not send ETH.
+
+    This call
+
+    > contract.withdraw(amount)
+
+    is equivalent to this call in the classic contract:
+
+    > contract.transact({'from': eth.defaultAccount, 'value': 0, ...}).withdraw(amount)
+    '''
+    def __getattr__(self, attr):
+        # If method_name.constant is true in the abi, then call by default
+        call_by_default = \
+                [m['constant'] if 'constant' in m.keys() else False \
+                    for m in self._classic_contract.abi if \
+                        m['type'] != 'constructor' and m['name'] == attr][0]
+        return ImplicitMethod(self._classic_contract, attr, call_by_default)
+
+class ImplicitMethod(ConciseMethod):
+    def __init__(self, contract, function, call_by_default):
+        self.call_by_default = call_by_default
+        super().__init__(contract, function)
+
+    def _prepared_function(self, **kwargs):
+        # Modifier is not provided and method is not constant/pure
+        # then do a transaction instead assuming default conditions
+        # Defaults are: from coinbase and no value sent
+        if not kwargs and not self.call_by_default:
+            default_account = self._contract.web3.eth.defaultAccount
+            return super()._prepared_function(transact={"from": default_account, "value": 0})
```

Comment:
I think that `value` defaults to zero if not included in the transaction dict which suggests it shouldn't be included here.

Reply:
:+1: so just `from` is needed?